### PR TITLE
chore: simplify alerting data model to leverage a single class 

### PR DESCRIPTION
**Pull request by bkyryliuk**
### SUMMARY
Simplifies alerting data models and merges validator and observer objects into the alert:
* moves observers and validators into alert
* moves observer and validator views into alert view
* migration includes object conversion into a new model for both downgrade and upgrade

### TEST PLAN
- [x] unit tests
- [x] testing on staging
deployed, migrated data & tested existing and new alerts on the dropbox staging env

local upgrade / downgrade
- [x] sqllite
- [x] mysql
- [ ] postgres

### CODE REVIEW

File: superset/migrations/versions/af30ca79208f_collapse_alerting_models_into_a_single_.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    for a in alerts:
+        if a.sql_observer:
+            a.sql = a.sql_observer[0].sql
+            a.database_id = a.sql_observer[0].database_id
+        if a.validators:
+            a.validator_type = a.validators[0].validator_type
+            a.validator_config = a.validators[0].config
+    session.commit()
+
+    # op.alter_column('alerts', 'database_id', server_default=None)
```
Comment (dpgaspar) [inline superset/migrations/versions/af30ca79208f_collapse_alerting_models_into_a_single_.py:142]: remove?

File: superset/views/alerts.py [hunk 2]
```diff
@@ -248,5 +170,12 @@ def pre_add(self, item: "AlertModelView") -> None:
         if not croniter.is_valid(item.crontab):
             raise SupersetException("Invalid crontab format")
 
+        item.validator_type = item.validator_type.lower()
+        check_validator(item.validator_type, item.validator_config)
+
+    def pre_update(self, item: "AlertModelView") -> None:
+        item.validator_type = item.validator_type.lower()
+        check_validator(item.validator_type, item.validator_config)
```
Comment (dpgaspar) [inline superset/views/alerts.py:178]: It brings a better used experience to use Flask-WTF validators instead: https://wtforms.readthedocs.io/en/2.3.x/validators/#module-wtforms.validators. This way the user get's alerted about a validation fail while still on the form and not after submitting

File: tests/alerts_tests.py [hunk 3]
```diff
@@ -165,41 +148,38 @@ def test_alert_observer(setup_database):
...
+    alert9 = create_alert(db_session, "SELECT {{ 2 }}")
+    observe(alert9.id, db_session)
+    assert alert9.observations[-1].value == 2.0
+    assert alert9.observations[-1].error_msg is None
 
 
 @patch("superset.tasks.schedules.deliver_alert")
 def test_evaluate_alert(mock_deliver_alert, setup_database):
-    dbsession = setup_database
+    db_session = setup_database
```
Comment (dpgaspar) [inline tests/alerts_tests.py:164]: sort of related, but looking at `setup_database` does it make sense at the cleanup to `DROP TABLE test_table` ?

**Review by dpgaspar — APPROVED** [review]
LGTM, left a couple of optional comments

File: tests/alerts_tests.py (hunk 3 above)
Comment (bkyryliuk) [reply tests/alerts_tests.py:164]: yes !

File: superset/views/alerts.py (hunk 2 above)
Comment (bkyryliuk) [reply superset/views/alerts.py:178]: this code will be removed in a couple of weeks as I understand, will keep it this way for now.

**Comment by bkyryliuk** [comment]
> LGTM, left a couple of optional comments

thanks for the review!

**Pull request merged**